### PR TITLE
Overdrive uses http helper

### PR DESCRIPTION
**Pull request by leonardr**
This branch fixes a problem where the Overdrive API was not properly using the util.http.HTTP class to make POST requests. To make sure the problem has been fixed I created a MockOverdriveAPI class and for the first time wrote tests for the procedure by which Overdrive creates and maintains its bearer token, as well as the various failures that might happen.

I also created a MockThreeMAPI class and wrote some basic tests for it. The real tests of the 3M API will be in the circulation manager.

### CODE REVIEW

File: threem.py [hunk 1]
```diff
@@ -131,13 +131,16 @@ def signature(self, method, path):
         signature = base64.b64encode(digest)
         return signature, now
 
-    def request(self, path, body=None, method="GET", identifier=None,
-                max_age=None):
+    def full_url(self, path):
```
Comment (leonardr) [inline threem.py]: This is unrelated--it's needed by the circulation branch I'm working on to get 3M using this system.

**Comment by aslagle** [comment]
👍

**Comment by aslagle** [comment]
:+1:
